modules/nextflow/src/test/groovy/nextflow/k8s/client/K8sClientTest.groovy

```diff
@@ -845,4 +845,121 @@ class K8sClientTest extends Specification {
         e.message == 'K8s pod configuration failed -- secrets "my-env" not found'
 
     }
+
+    def 'client should throw process exception when initContainer fails' () {
+        // Addresses bug 2428, which is when an initContainer exists and fails
+        // leaving the containerStatuses in "PodInitializing", but the pod
+        // phase is "Failed".
+        def JSON = '''
+            {
+                "apiVersion": "v1",
+                "kind": "Pod",
+                "metadata": {
+                    "creationTimestamp": "2021-11-03T21:37:45Z",
+                    "labels": {
+                        "app": "nextflow",
+                        "magic-version": "v7.5.3",
+                        "processName": "some_special_magic",
+                        "runName": "lonely_tuckerman",
+                        "sessionId": "uuid-41ef6f42-bab5-438f-83e4-85e91a6386ea",
+                        "taskName": "some_special_magic_1"
+                    },
+                    "name": "nf-45d62fd8462ba390bacfa20ad9065bfe",
+                    "namespace": "default",
+                    "resourceVersion": "90692",
+                    "uid": "280bf29c-4c14-4451-a300-080760ff7a0e"
+                },
+                "status": {
+                    "conditions": [
+                        {
+                            "lastProbeTime": null,
+                            "lastTransitionTime": "2021-11-03T21:37:45Z",
+                            "message": "containers with incomplete status: [magic-init]",
+                            "reason": "ContainersNotInitialized",
+                            "status": "False",
+                            "type": "Initialized"
+                        },
+                        {
+                            "lastProbeTime": null,
+                            "lastTransitionTime": "2021-11-03T21:37:45Z",
+                            "message": "containers with unready status: [nf-45d62fd8462ba390bacfa20ad9065bfe]",
+                            "reason": "ContainersNotReady",
+                            "status": "False",
+                            "type": "Ready"
+                        },
+                        {
+                            "lastProbeTime": null,
+                            "lastTransitionTime": "2021-11-03T21:37:45Z",
+                            "message": "containers with unready status: [nf-45d62fd8462ba390bacfa20ad9065bfe]",
+                            "reason": "ContainersNotReady",
+                            "status": "False",
+                            "type": "ContainersReady"
+                        },
+                        {
+                            "lastProbeTime": null,
+                            "lastTransitionTime": "2021-11-03T21:37:45Z",
+                            "status": "True",
+                            "type": "PodScheduled"
+                        }
+                    ],
+                    "containerStatuses": [
```

Comment:
Maybe the `conditions` can be removed to simplify the test sample

Reply:
Sure, just removed them.

Reply:
Awesome. Thanks Mike!